src/deviceTree.ts

```diff
@@ -12,6 +12,13 @@ export class DeviceTree implements vscode.TreeDataProvider<vscode.TreeItem> {
     public readonly onDidChangeTreeData: vscode.Event<vscode.TreeItem | undefined> = this._onDidChangeTreeData.event;
 
     constructor(private context: vscode.ExtensionContext) {
+        let treeViewAutoRefreshEnable = Utility.getConfig<boolean>(Constants.TreeViewAutoRefreshEnableKey);
+        if (treeViewAutoRefreshEnable) {

```

Comment:
Think about the scenario: The interval is set as 5 seconds, while the tree view takes 8 seconds to render.
Same with your concern, would better to put this logic into `getChildren()`, `setInterval` after tree view is rendered, 
If the logic is in `getChildren()`, we also need to `clearTimeout` for previous `setInterval`.